Panic in storage

I first caused the issues running the `select mean(usage_idle) from telegraf..cpu where time > now() - 1m` query through the transpiler and ifqld. Running the transpiled query spec by hand causes the issues as well. Although its not been consistently the same query that causes the issue.


```
panic: runtime error: slice bounds out of range
goroutine 7848 [running]:
github.com/influxdata/influxdb/services/storage.(*indexSeriesCursor).Next(0xc4457cc000, 0x1)
	/home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/services/storage/series_cursor.go:174 +0x72b
github.com/influxdata/influxdb/services/storage.groupBySort(0xc460d7d080, 0xe7a988, 0x133ae28, 0x2)
	/home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/services/storage/group_resultset.go:248 +0x100
github.com/influxdata/influxdb/services/storage.(*groupResultSet).sort(0xc460d7d080, 0x0, 0x0, 0x0)
	/home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/services/storage/group_resultset.go:122 +0xcb
github.com/influxdata/influxdb/services/storage.newGroupResultSet(0xe72ec0, 0xc443af57d0, 0xc43885d900, 0xc45b89e7c0, 0xc4200c22b8)
	/home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/services/storage/group_resultset.go:85 +0x2ba
github.com/influxdata/influxdb/services/storage.(*Store).GroupRead(0xc44a2c2020, 0xe72ec0, 0xc443af57d0, 0xc43885d900, 0x0, 0x0, 0x0)
	/home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/services/storage/store.go:159 +0x297
github.com/influxdata/influxdb/services/storage.(*rpcService).handleGroupRead(0xc44a2c2060, 0xe72ec0, 0xc443af57d0, 0xc43885d900, 0xc45a5d7d70, 0x0, 0x0)
	/home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/services/storage/rpc_service.go:188 +0x6b
github.com/influxdata/influxdb/services/storage.(*rpcService).Read(0xc44a2c2060, 0xc43885d900, 0xe73180, 0xc457c7b3b0, 0x0, 0x0)
	/home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/services/storage/rpc_service.go:134 +0xf06
github.com/influxdata/influxdb/services/storage._Storage_Read_Handler(0xcf9f00, 0xc44a2c2060, 0xe71380, 0xc45b89e680, 0xcb8287dd9686c01, 0xc00000000000000)
	/home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/services/storage/storage.yarpc.go:134 +0x10e
github.com/influxdata/influxdb/vendor/github.com/influxdata/yarpc.(*Server).handleStreamingRPC(0xc455618b10, 0xc4556c0340, 0xc44a2c2080, 0x130a7a0)
	/home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/vendor/github.com/influxdata/yarpc/server.go:220 +0xfe
github.com/influxdata/influxdb/vendor/github.com/influxdata/yarpc.(*Server).handleStream(0xc455618b10, 0xc4556c0340)
	/home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/vendor/github.com/influxdata/yarpc/server.go:199 +0x1c6
created by github.com/influxdata/influxdb/vendor/github.com/influxdata/yarpc.(*Server).serveSession
	/home/nathanielc/projects/influxdb/src/github.com/influxdata/influxdb/vendor/github.com/influxdata/yarpc/server.go:166 +0x4a
```

--- a/services/storage/group_resultset.go
+++ b/services/storage/group_resultset.go
@@ -186,18 +186,20 @@ func groupNoneSort(g *groupResultSet) (int, error) {
 
 	n := 0
 	row := cur.Next()
-	g.km.setTags(row.tags)
+	if row != nil {
+		g.km.setTags(row.tags)
 
-	allTime := g.req.Hints.HintSchemaAllTime()
+		allTime := g.req.Hints.HintSchemaAllTime()
 
-	for {
-		n++
-		row = cur.Next()
-		if row == nil {
-			break
-		}
-		if allTime || g.seriesHasPoints(row) {
-			g.km.mergeTagKeys(row.tags)
+		for {
+			n++
+			row = cur.Next()
+			if row == nil {
+				break
+			}
+			if allTime || g.seriesHasPoints(row) {
+				g.km.mergeTagKeys(row.tags)
+			}
 		}
 	}
 

--- a/services/storage/series_cursor.go
+++ b/services/storage/series_cursor.go
@@ -116,6 +116,9 @@ func newIndexSeriesCursor(ctx context.Context, predicate *Predicate, shards []*t
 
 			p.fields = extractFields(fi)
 			fi.Close()
+			if len(p.fields) == 0 {
+				goto CLEANUP
+			}
 			return p, nil
 		}
 	}
@@ -150,31 +153,33 @@ func (c *indexSeriesCursor) Next() *seriesRow {
 		return nil
 	}
 
-RETRY:
-	if len(c.nf) == 0 {
-		// next series key
-		sr, err := c.sqry.Next()
-		if err != nil {
-			c.err = err
-			c.Close()
-			return nil
-		} else if sr == nil {
-			c.Close()
-			return nil
+	for {
+		if len(c.nf) == 0 {
+			// next series key
+			sr, err := c.sqry.Next()
+			if err != nil {
+				c.err = err
+				c.Close()
+				return nil
+			} else if sr == nil {
+				c.Close()
+				return nil
+			}
+
+			c.row.name = sr.Name
+			c.row.stags = sr.Tags
+			c.tags = copyTags(c.tags, sr.Tags)
+			c.tags.Set(measurementKey, sr.Name)
+
+			c.nf = c.fields[string(sr.Name)]
+			// c.nf may be nil if there are no fields
+		} else {
+			c.row.field, c.nf = c.nf[0], c.nf[1:]
+
+			if c.measurementCond == nil || evalExprBool(c.measurementCond, c) {
+				break
+			}
 		}
-
-		c.row.name = sr.Name
-		c.row.stags = sr.Tags
-		c.tags = copyTags(c.tags, sr.Tags)
-		c.tags.Set(measurementKey, sr.Name)
-
-		c.nf = c.fields[string(sr.Name)]
-	}
-
-	c.row.field, c.nf = c.nf[0], c.nf[1:]
-
-	if c.measurementCond != nil && !evalExprBool(c.measurementCond, c) {
-		goto RETRY
 	}
 
 	c.tags.Set(fieldKey, c.row.field.nb)

--- a/services/storage/store.go
+++ b/services/storage/store.go
@@ -153,7 +153,11 @@ func (s *Store) GroupRead(ctx context.Context, req *ReadRequest) (*groupResultSe
 	req.TimestampRange.End = end
 
 	newCursor := func() (seriesCursor, error) {
-		return newIndexSeriesCursor(ctx, req.Predicate, shards)
+		cur, err := newIndexSeriesCursor(ctx, req.Predicate, shards)
+		if cur == nil || err != nil {
+			return nil, err
+		}
+		return cur, nil
 	}
 
 	return newGroupResultSet(ctx, req, newCursor), nil

